### ISSUE TITLE
sql: support inet data types

### ISSUE THREAD

**Issue by abh** [cockroachdb/cockroach#6981]
The inet and cidr data types in postgresql are very powerful for applications that deal with IP addresses; in particular IPv6 ones.

https://www.postgresql.org/docs/current/static/datatype-net-types.html
https://www.postgresql.org/docs/current/static/functions-net.html

**Comment by jordanlewis** [comment]
This is required by a popular app that runs on Postgres - Mastodon.

**Comment by nickjacques** [comment]
+1, [this is required for PowerDNS schema compatibility when using the postgresql backend.](https://github.com/PowerDNS/pdns/blob/master/modules/gpgsqlbackend/schema.pgsql.sql#L40)

**Comment by dianasaur323** [comment]
adding @nstewart to the chain

**Comment by aaomidi** [comment]
This is absolutely required for any modern day application. It being open for more than a year with barely any updates is worrisome.

**Comment by nstewart** [comment]
This is on our radar and is scheduled to be part of the 1.2 release (mid-2018). We expect it to be in an alpha version of CockroachDB much earlier than that.

**Comment by vivekmenezes** [comment]
We are not adding CIDR support now until we find a need for it. Thanks!

**Comment by aaomidi** [comment]
What defines a need?

**Comment by lego** [comment]
In the case of CIDR, if people bring up major projects that make use of that column specifically then I'd be more than happy to spend some time adding it. To my knowledge, CIDR is strictly less useful than INET as it seems to be a subset of INET. I'll be closing this as I've added the INET type and it's built-ins, but I created an issue to track the CIDR type, #18846.

I also didn't add many of the operators postgres supports yet so there's another issue to track that, #18847. I didn't find the operators were in common use so I just didn't get around to it yet.
